transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStreamServiceImpl.java

```diff
@@ -122,4 +127,18 @@ public void onServerRemoved(final MemberId memberId) {
   public ClientStreamer<M> streamer() {
     return this;
   }
+
+  @Override
+  public Optional<ClientStream<M>> streamFor(final ClientStreamId id) {
+    // mapping to itself is necessary to cast from impl to interface type
+    return registry.getClient(id).map(s -> s);
```

Comment:
:x: `registry` is not thread safe as far as I know. Right now this is only used in tests. But to be safe, it would be better to execute this in actor, and return an `ActorFuture`.

Reply:
Good catch, I think I assumed it was because of the broker one.